registry/remote/auth/example_test.go

```diff
@@ -193,18 +185,11 @@ func ExampleClient_Do_basicAuth() {
 // including using cache, setting user agent and configuring OAuth2.
 func ExampleClient_Do_clientConfigurations() {
 	client := &auth.Client{
-		Credential: func(ctx context.Context, reg string) (auth.Credential, error) {
-			switch reg {
-			// expectedHostAddress is of form ipaddr:port
-			case expectedHostAddress:
-				return auth.Credential{
-					Username: username,
-					Password: password,
-				}, nil
-			default:
-				return auth.EmptyCredential, fmt.Errorf("credential not found for %v", reg)
```

Comment:
Just realized that this returns an error when credential is not found for registry. Do you think we should also return error in the utility method for this case? /cc @qweeah

Reply:
Yes, we should when there is an registry host name mismatch

Reply:
Added the error.